packages/coreneuron/package.py

```diff
@@ -173,12 +174,13 @@ def install(self, spec, prefix):
 
             if mech_set:
                 options.extend(['-DADDITIONAL_MECHPATH=%s' % (modlib_dir)])
-
-            cmake('..', *options)
             self.profiling_wrapper_on()
-            make()
-            make('install')
-            self.profiling_wrapper_off()
+        return options
+#            cmake('..', *options)
+#            self.profiling_wrapper_on()
```

Comment:
@fouriaux : thank you for changes!

Minor comments / good to have things as part of this refactoring : 

* `-DCMAKE_INSTALL_PREFIX` is redundant now (set by CMakePackage)

* Could you remove commented code and add `before` / `after` decorators? For example,
you have to just add:

    ```
    @run_before('install')
    def profiling_wrapper_on(self):
    ....

    @run_after('install')
     def profiling_wrapper_off(self):
     ....
     ```
this will allow to build `+profile` variants with TAU.  See documentation [here](http://spack.readthedocs.io/en/latest/packaging_guide.html#build-time-tests).

* I believe this may not work with `setup` because we are setting `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` compilers explicitly. The reason is that Spack has it's own wrapper for compilers and we can't use those for `setup` workflow. Good news is that the package don't need to set compilers at all because Spack sets environmental variables `CC`, `CXX` etc. So we should remove setting compilers using `CMAKE_C_COMPILER` / `CMAKE_CXX_COMPILER`.  And for the cases where we need to set those (mentioned below), we should use `env['CC']`, `env['CXX']` etc. See docs [here](http://spack.readthedocs.io/en/latest/packaging_guide.html#mpi-support-in-spack). 

below code should be replace with use of `env`
```
            if spec.satisfies('+profile'):
                c_compiler = 'tau_cc'
                cxx_compiler = 'tau_cxx'
            elif 'bgq' in self.spec.architecture and spec.satisfies('+mpi'):
                # for bg-q, our cmake is not setup properly
                c_compiler = spec['mpi'].mpicc
                cxx_compiler = spec['mpi'].mpicxx
```